deltacat/compute/compactor_v2/utils/exception_handler.py

```diff
@@ -0,0 +1,126 @@
+import botocore
+from typing import Optional
+from ray.exceptions import (
+    RayError,
+    RuntimeEnvSetupError,
+    WorkerCrashedError,
+    NodeDiedError,
+    OutOfMemoryError,
+)
+from pyarrow.lib import ArrowException, ArrowInvalid, ArrowCapacityError
+from deltacat.exceptions import (
+    DependencyPyarrowError,
+    DependencyRayRuntimeSetupError,
+    DependencyPyarrowInvalidError,
+    DependencyPyarrowCapacityError,
+    DependencyDaftError,
+    GeneralAssertionError,
+    RetryableTimeoutError,
+    DependencyRayWorkerDiedError,
+    DependencyRayOutOfMemoryError,
+    DependencyRayError,
+)
+
+
+RAY_TASK_RETRYABLE_TIMEOUT_ERROR_CODES = (
+    botocore.exceptions.ConnectionError,
+    botocore.exceptions.HTTPClientError,
+    ConnectionError,
+    TimeoutError,
+)
+
+
+def parametrized(dec):
+    def layer(*args, **kwargs):
+        def repl(f):
+            return dec(f, *args, **kwargs)
+
+        return repl
+
+    return layer
+
+
+@parametrized
+def handle_exception(func, task_id):
+    def handle_compaction_exception(*args, **kwargs):
+        try:
+            res = func(*args, **kwargs)
```

Comment:
```suggestion
            return func(*args, **kwargs)
```

Unreferenced variable error will occur if `handle_compaction_step_exception` doesn't raise anything. Does it make sense to return res within the try itself?